### PR TITLE
Feature / comment-popup

### PR DESCRIPTION
**Pull request by munkk**

### CODE REVIEW

File: UI/src/app/app.module.ts [hunk 1]
```diff
@@ -16,31 +16,32 @@ import {MatButtonModule} from '@angular/material/button';
 import {MatCardModule} from '@angular/material/card';
 import {MatExpansionModule} from '@angular/material/expansion';
 import { MatTableModule } from '@angular/material/table';
+import {MatInputModule, MatFormFieldModule} from '@angular/material';
```
Comment (etokareva) [inline UI/src/app/app.module.ts]: Try to use the same style of indentations everywhere

File: UI/src/app/pages/mapping/components/dialog/dialog.component.html [hunk 2]
```diff
@@ -0,0 +1,88 @@
+<mat-card class="dialog">
+  <mat-list #shoes>
```
Comment (etokareva) [inline UI/src/app/pages/mapping/components/dialog/dialog.component.html]: why shoes?

File: UI/src/app/pages/mapping/components/dialog/dialog.component.scss [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    display: flex;
+    flex-direction: column;
+    align-items: flex-end;
+}
+
+.comment-read {
+    &-head {
+        width: 100%;
+        display: flex;
+        flex-direction: row;
```
Comment (etokareva) [inline UI/src/app/pages/mapping/components/dialog/dialog.component.scss]: Does direction equal to 'row' by default?

File: UI/src/app/pages/mapping/components/panel/panel-table-column/panel-table-column.component.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { Component, OnInit, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-panel-table-column',
+  templateUrl: './panel-table-column.component.html',
+  styleUrls: ['./panel-table-column.component.scss']
+})
+export class PanelTableColumnComponent implements OnInit {
```
Comment (etokareva) [inline UI/src/app/pages/mapping/components/panel/panel-table-column/panel-table-column.component.ts]: no need in emty OnInit

File: UI/src/app/pages/mapping/components/dialog/dialog.component.html [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    <ng-container *ngFor="let comment of comments">
+      <ng-template 
+      [ngTemplateOutlet]="loadTemplate(comment)" 
+      [ngTemplateOutletContext]="{ $implicit: comment}">
+      </ng-template>
+    </ng-container>
+  </mat-list>
+
+  <input [(ngModel)]="value" [disabled]="isDisabled()" placeholder="Input comment" class="input input-add" (keyup.enter)="add()">
```
Comment (etokareva) [inline UI/src/app/pages/mapping/components/dialog/dialog.component.html]: prefer not to use several classes.
e.g. 'input-add' would be enough here.
Inside of scss you can use `@extend .classname;`

```scss
.input {
 // some styles for input
}
.input-add {
   @extend .input; //adds rules from .input to .input-add
   //some additional styles 
}
```

File: UI/src/app/pages/mapping/components/dialog/dialog.component.html [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        <p class="comment-read-text">
+          {{comment.text}}
+        </p>
+      </div>
+    </div>
+  </mat-list-item>
+</ng-template>
+
+<ng-template #editTemplate let-comment>
+  <mat-list-item class="list-item-edit">
```
Comment (etokareva) [inline UI/src/app/pages/mapping/components/dialog/dialog.component.html]: As soon as our styles are incapsulated inside of the component we could use simpler class names  or even use tags if it's possible

File: UI/src/app/pages/mapping/components/dialog/dialog.component.ts [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+export class Comment {
+  id: number;
+  text: string;
+  date: Date;
+  hasBeenEdited: false;
+
+  constructor(id: number, text: string) {
+    this.id = id;
```
Comment (etokareva) [inline UI/src/app/pages/mapping/components/dialog/dialog.component.ts]: no need in ```this.id = id; this.text = text;``` inside of the constructor.
You can read about it [there](https://kendaleiv.com/typescript-constructor-assignment-public-and-private-keywords/)

File: UI/src/app/pages/mapping/components/panel/panel-table/panel-table.component.ts [hunk 8]
```diff
@@ -0,0 +1,67 @@
+
+import { Component, OnInit, Input, ViewChild, ElementRef, Injector, ChangeDetectionStrategy, ChangeDetectorRef } from '@angular/core';
+import { Store, select } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { CommentsService } from 'src/app/pages/mapping/services/comments.service';
+import { DragService } from 'src/app/pages/mapping/services/drag.service';
+import { Overlay, CdkOverlayOrigin, OverlayRef, OverlayConfig } from '@angular/cdk/overlay';
```
Comment (etokareva) [inline UI/src/app/pages/mapping/components/panel/panel-table/panel-table.component.ts]: try to follow the next order of importing:
1) foreign libs
2) your own scripts

File: UI/src/app/pages/mapping/components/area/area.component.ts [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Collapsed, Expanded, Linked } from 'src/app/pages/mapping/store/actions/common.actions';
+
+@Component({
+  selector: 'app-area',
+  templateUrl: './area.component.html',
+  styleUrls: ['./area.component.scss']
+})
+export class AreaComponent implements OnInit {
```
Comment (etokareva) [inline UI/src/app/pages/mapping/components/area/area.component.ts]: empty OnInit?

File: UI/src/app/pages/mapping/mapping.component.html [hunk 10]
```diff
@@ -0,0 +1,38 @@
+<div class="main">
+    
+    <app-area [area]="'source'">
```
Comment (etokareva) [inline UI/src/app/pages/mapping/mapping.component.html]: why do we use such type of binding for constant string?

File: UI/src/app/pages/mapping/store/reducers/comments.reducer.ts [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+export interface Panel {
+  [rowName: string]: Row;
+}
+
+export interface Row {
+  comments: IComment[];
+}
+
+export interface State {
+    'source': Area;
```
Comment (etokareva) [inline UI/src/app/pages/mapping/store/reducers/comments.reducer.ts]: State is an object, simple objects in JS can have only string keys.
So when you use quotes - it's not a mistake. But try to use the same style of coding everywhere

File: UI/src/app/pages/mapping/store/reducers/common.reducer.ts [hunk 12]
```diff
@@ -1,15 +1,15 @@
 import { Action } from '@ngrx/store';
-import { ActionTypes } from 'src/app/store/actions/common.actions';
+import { ActionTypes, CommonAction } from 'src/app/pages/mapping/store/actions/common.actions';
 
 export interface State {
-    hint: string;
+    convasHeight: number;
```
Comment (etokareva) [inline UI/src/app/pages/mapping/store/reducers/common.reducer.ts]: canvasHeight

File: UI/src/styles.scss [hunk 13]
```diff
@@ -8,3 +8,32 @@ body {
...
+}
+
+@-webkit-keyframes gb__a{0%{opacity:0}50%{opacity:1}}
+@keyframes gb__a{0%{opacity:0}50%{opacity:1}}
+
+.mat-expansion-panel-body {
+    padding: 0 !important;
+}
+
+.custom-backdrop {
```
Comment (etokareva) [inline UI/src/styles.scss]: get rid of empty class

File: UI/src/app/pages/mapping/components/dialog/dialog.component.ts (hunk 7 above)
Comment (munkk) [reply UI/src/app/pages/mapping/components/dialog/dialog.component.ts]: i thought explicit version would be more readable)

**Pull request closed without merging**